Mathlib/Data/Matrix/Defs.lean

```diff
@@ -497,4 +497,61 @@ abbrev subDownLeft {d u l r : Nat} (A : Matrix (Fin (u + d)) (Fin (l + r)) α) :
     Matrix (Fin d) (Fin l) α :=
   subDown (subLeft A)
 
+section RowCol
+
+/-- For an `m × n` `α`-matrix `A`, `A.row i` is the `i`th row of `A` as a vector in `n → α`.
+`A.row` is defeq to `A`, but explicitly refers to the 'row function` of `A`
+while avoiding defeq abuse and noisy eta-expansions,
+such as in expressions like `Set.Injective A.row` and `Set.range A.row`.
+(Note 03-15-2025 : the identifier `Matrix.row` used to refer to a matrix with constant rows;
+this is now called `Matrix.rowConst`) -/
+def row (A : Matrix m n α) : m → n → α := A
+
+/-- For an `m × n` `α`-matrix `A`, `A.col j` is the `j`th column of `A` as a vector in `m → α`.
+`A.col` is defeq to `Aᵀ`, but refers to the 'column function' of `A`
+while avoiding defeq abuse and noisy eta-expansions
+(and without the simplifier unfolding transposes) in expressions like `Set.Injective A.col`
+and `Set.range A.col`.
+(Note 03-15-2025 : the identifier `Matrix.col` used to refer to a matrix with constant columns;
+this is now called `Matrix.colConst`) -/
+def col (A : Matrix m n α) : n → m → α := Aᵀ
+
+lemma row_eq_self (A : Matrix m n α) : A.row = A := rfl
```

Comment:
```suggestion
lemma row_eq_self (A : Matrix m n α) : A.row = of.symm A := rfl
```
etc, to avoid inconsistent types. 

I think @kim-em has a linter in the works that would reject what you currently have for this reason.